### PR TITLE
Clearer error message for simple_bind failure

### PR DESCRIPTION
**Pull request by reminisce**

### CODE REVIEW

File: python/mxnet/symbol.py [hunk 1]
```diff
@@ -1368,11 +1368,12 @@ def simple_bind(self, ctx, grad_req='write', type_dict=None, group2ctx=None,
...
                                                  shared_exec_handle,
                                                  ctypes.byref(exe_handle)))
-        except MXNetError:
-            print("simple_bind error. Arguments:")
+        except MXNetError as e:
+            error_msg = "simple_bind error. Arguments:"
             for k, v in kwargs.items():
-                print("  %s: %s" % (k, v))
-            raise RuntimeError('simple_bind failed')
+                error_msg += "  %s: %s" % (k, v)
```
Comment (piiswrong) [inline python/mxnet/symbol.py]: \n?

**Comment by piiswrong** [comment]
what does the error look like?

**Comment by reminisce** [comment]
With this script:
```python
def test_error_msg():
    x = mx.sym.Variable('x')
    y = mx.sym.Variable('y')
    sym = mx.sym.dot(x, y)
    exe = sym.simple_bind(ctx=mx.test_utils.default_context(), x=(2, 0), y=(2, 0))
    exe.forward()

test_error_msg()
```
Backend error message:
```
[16:12:47] /Users/jwum/Development/mxnet/dmlc-core/include/dmlc/logging.h:304: [16:12:47] src/executor/graph_executor.cc:380: InferShape pass cannot decide shapes for the following arguments (0s means unknown dimensions). Please consider providing them as inputs:
x: (2,0), y: (2,0), 

Stack trace returned 6 entries:
[bt] (0) 0   libmxnet.so                         0x0000000107eb2275 _ZN4dmlc15LogMessageFatalD2Ev + 37
[bt] (1) 1   libmxnet.so                         0x0000000108501b12 _ZN5mxnet4exec21HandleInferShapeErrorEmRKN4nnvm12IndexedGraphERKNSt3__16vectorINS1_6TShapeENS5_9allocatorIS7_EEEE + 882
[bt] (2) 2   libmxnet.so                         0x000000010850b6ba _ZN5mxnet4exec13GraphExecutor4InitEN4nnvm6SymbolERKNS_7ContextERKNSt3__13mapINS7_12basic_stringIcNS7_11char_traitsIcEENS7_9allocatorIcEEEES4_NS7_4lessISE_EENSC_INS7_4pairIKSE_S4_EEEEEERKNS7_6vectorIS4_NSC_IS4_EEEESS_SS_RKNS7_13unordered_mapISE_NS2_6TShapeENS7_4hashISE_EENS7_8equal_toISE_EENSC_INSH_ISI_SU_EEEEEERKNST_ISE_iSW_SY_NSC_INSH_ISI_iEEEEEERKNSO_INS_9OpReqTypeENSC_IS19_EEEERKNS7_13unordered_setISE_SW_SY_NSC_ISE_EEEEPNSO_INS_7NDArrayENSC_IS1J_EEEES1M_S1M_PNST_ISE_S1J_SW_SY_NSC_INSH_ISI_S1J_EEEEEEPNS_8ExecutorERKNST_INS2_9NodeEntryES1J_NS2_13NodeEntryHashENS2_14NodeEntryEqualENSC_INSH_IKS1T_S1J_EEEEEE + 1482
[bt] (3) 3   libmxnet.so                         0x000000010850e0d1 _ZN5mxnet8Executor10SimpleBindEN4nnvm6SymbolERKNS_7ContextERKNSt3__13mapINS6_12basic_stringIcNS6_11char_traitsIcEENS6_9allocatorIcEEEES3_NS6_4lessISD_EENSB_INS6_4pairIKSD_S3_EEEEEERKNS6_6vectorIS3_NSB_IS3_EEEESR_SR_RKNS6_13unordered_mapISD_NS1_6TShapeENS6_4hashISD_EENS6_8equal_toISD_EENSB_INSG_ISH_ST_EEEEEERKNSS_ISD_iSV_SX_NSB_INSG_ISH_iEEEEEERKNSN_INS_9OpReqTypeENSB_IS18_EEEERKNS6_13unordered_setISD_SV_SX_NSB_ISD_EEEEPNSN_INS_7NDArrayENSB_IS1I_EEEES1L_S1L_PNSS_ISD_S1I_SV_SX_NSB_INSG_ISH_S1I_EEEEEEPS0_ + 673
[bt] (4) 4   libmxnet.so                         0x00000001084bcc3a MXExecutorSimpleBind + 8458
[bt] (5) 5   _ctypes.so                          0x0000000100750677 ffi_call_unix64 + 79
```

Front-end error message:
```
Traceback (most recent call last):
  File "/Users/jwum/Development/sandbox/test_simple_bind.py", line 96, in <module>
    test_error_msg()
  File "/Users/jwum/Development/sandbox/test_simple_bind.py", line 93, in test_error_msg
    exe = sym.simple_bind(ctx=mx.test_utils.default_context(), x=(2, 0), y=(2, 0))
  File "/Users/jwum/anaconda/lib/python2.7/site-packages/mxnet-0.10.1-py2.7.egg/mxnet/symbol.py", line 1376, in simple_bind
    raise RuntimeError(error_msg)
RuntimeError: simple_bind error. Arguments:
y: (2, 0)
x: (2, 0)
Exception error is: [16:12:47] src/executor/graph_executor.cc:380: InferShape pass cannot decide shapes for the following arguments (0s means unknown dimensions). Please consider providing them as inputs:
x: (2,0), y: (2,0), 
```

File: python/mxnet/symbol.py [hunk 2]
```diff
@@ -1368,11 +1368,12 @@ def simple_bind(self, ctx, grad_req='write', type_dict=None, group2ctx=None,
...
                                                  ctypes.byref(exe_handle)))
-        except MXNetError:
-            print("simple_bind error. Arguments:")
+        except MXNetError as e:
+            error_msg = "simple_bind error. Arguments:\n"
             for k, v in kwargs.items():
-                print("  %s: %s" % (k, v))
-            raise RuntimeError('simple_bind failed')
+                error_msg += "%s: %s\n" % (k, v)
+            error_msg += 'Exception error is: %s' % e
```
Comment (piiswrong) [inline python/mxnet/symbol.py]: remove Exception error is:

Comment (reminisce) [inline python/mxnet/symbol.py]: Done.

**Pull request merged**